### PR TITLE
Thiefbird

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 1]
```diff
@@ -93,93 +146,1019 @@ static DamageTable D_80C13644 = {
...
+    EnThiefbird* this = THIS;
+
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+void func_80C10984(EnThiefbird* this, s32 arg1) {
+    s32 i;
+    EnThiefbirdUnkStruct* ptr = &this->unk_3F0[0];
+
+    for (i = 0; i < 0x28; i++, ptr++) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:199]:
```suggestion
    for (i = 0; i < ARRAY_COUNT(this->unk_3F0); i++, ptr++) {
```

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 2]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+void EnThiefbird_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnThiefbird* this = THIS;
+    s32 i;
+    ColliderJntSphElementDim* dim;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_060061A0, &D_06000604, this->jointTable, this->morphTable, 17);
+    Collider_InitAndSetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit, this->colliderElements);
+
+    for (i = 0; i < 3; i++) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:168]:
```suggestion
    for (i = 0; i < ARRAY_COUNT(this->colliderElements); i++) {
```

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 3]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+            if (arg1 == 0) {
+                break;
+            }
+        }
+    }
+}
+
+s32 func_80C10B0C(EnThiefbird* this, GlobalContext* globalCtx) {
+    static UNK_TYPE D_80C13680[] = { &D_06004348, &D_06004B88, &D_060055E0 };
+    s32 itemFound = false;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:221]:
```suggestion
    s32 isItemFound = false;
```

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 4]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+    for (i = 0; i < phi_s2; i++) {
+        if (Rand_ZeroOne() < 0.5f) {
+            phi_s0_2--;
+        }
+    }
+
+    i = sp5C - phi_s0_2;
+    if (i) {}
+    sp5C = phi_s0_2 * 50;
+    sp98 = sp98 - sp5C;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:365]:
```suggestion
    sp98 -= sp5C;
```

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 5]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+
+    func_80C10DE8(sp74, phi_s0_2, ITEM00_RUPEE_PURPLE);
+    spA0 = sp98 / 20;
+    if (i < spA0) {
+        spA0 = i;
+    }
+    i -= spA0;
+    sp98 = sp98 - (spA0 * 20);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:373]:
```suggestion
    sp98 -= spA0 * 20;
```

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 6]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+
+    func_80C10DE8(sp74, spA0, ITEM00_RUPEE_RED);
+    phi_s2 = sp98 / 5;
+    if (i < phi_s2) {
+        phi_s2 = i;
+    }
+    i -= phi_s2;
+    sp98 = sp98 - (phi_s2 * 5);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:381]:
```suggestion
    sp98 -= phi_s2 * 5;
```

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 7]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+    } else {
+        if (Actor_XZDistanceToPoint(&this->actor, &this->actor.home.pos) > 300.0f) {
+            this->unk_192 = Actor_YawToPoint(&this->actor, &this->actor.home.pos);
+        }
+    }
+
+    if (!Math_SmoothStepToS(&this->actor.shape.rot.y, this->unk_192, 5, 0x300, 0x10) && (sp38 != 0) &&
+        (Rand_ZeroOne() < 0.1f)) {
+        s16 yaw = Actor_YawToPoint(&this->actor, &this->actor.home.pos) - this->actor.shape.rot.y;
+        if (yaw > 0) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:499]:
```suggestion
        s16 yaw = Actor_YawToPoint(&this->actor, &this->actor.home.pos) - this->actor.shape.rot.y;
        
        if (yaw > 0) {
```

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 8]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+    }
+
+    pitch = Actor_PitchBetweenActors(&this->actor, &player->actor);
+    pitch = CLAMP(pitch, -0x2800, 0x2800);
+    Math_SmoothStepToS(&this->actor.shape.rot.x, pitch, 4, 0x800, 0x80);
+    if (this->actor.bgCheckFlags & 8) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.wallYaw, 6, 0x1000, 0x100);
+    } else if (Actor_IsActorFacingLink(&this->actor, 0x3C00) || (this->actor.xzDistToPlayer > 120.0f)) {
+        s16 rot = BINANG_ROT180(this->actor.yawTowardsPlayer - player->actor.shape.rot.y);
+        if (rot > 0x4000) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:563]:
```suggestion
        s16 rot = BINANG_ROT180(this->actor.yawTowardsPlayer - player->actor.shape.rot.y);
        
        if (rot > 0x4000) {
```

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 9]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+void func_80C11DF0(EnThiefbird* this, GlobalContext* globalCtx) {
+    s32 i;
+    s32 j;
+
+    if (Math_ScaledStepToS(&this->actor.shape.rot.x, -0x8000, 0x800)) {
+        Math_ScaledStepToS(&this->unk_192, 0x1C00, 0x200);
+        this->actor.shape.rot.y += this->unk_192;
+    }
+
+    if ((this->actor.bgCheckFlags & 1) || (this->actor.floorHeight == -32000.0f)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:656]:
```suggestion
    if ((this->actor.bgCheckFlags & 1) || (this->actor.floorHeight == BGCHECK_Y_MIN)) {
```

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 10]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+    } else if (this->unk_3E8 == 0) {
+        this->unk_192 = BINANG_ROT180(this->actor.yawTowardsPlayer);
+    } else {
+        this->unk_192 = Actor_YawToPoint(&this->actor, &D_80C13920);
+        this->unk_190 = Actor_PitchToPoint(&this->actor, &D_80C13920);
+    }
+
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->unk_192, 6, 0x1000, 0x100);
+    Math_SmoothStepToS(&this->actor.shape.rot.x, this->unk_190, 5, 0x100, 0x10);
+    sp2C = (this->unk_18E * 153.6f) + 2048.0f;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:745]:
```suggestion
    sp2C = (this->unk_18E * 153.6f) + 0x800;
```
May need `(f32)0x800`

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 11]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+        if (this->actor.colChkInfo.damageEffect == 3) {
+            func_80C11454(this);
+        } else if (this->actor.colChkInfo.damageEffect == 4) {
+            this->unk_18C = 20;
+            this->unk_3D8 = 0.5f;
+            this->unk_3D4 = 4.0f;
+            if (i != ARRAY_COUNT(this->colliderElements)) {
+                sph = &this->collider.elements[i];
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG, sph->info.bumper.hitPos.x,
+                            sph->info.bumper.hitPos.y, sph->info.bumper.hitPos.z, 0, 0, 0, 4);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:950]:
```suggestion
                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG, sph->info.bumper.hitPos.x,
                            sph->info.bumper.hitPos.y, sph->info.bumper.hitPos.z, 0, 0, 0, CLEAR_TAG_LARGE_LIGHT_RAYS);
```

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 12]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+            Math_Vec3f_Sum(&ptr->unk_0C, &D_80C1368C, &ptr->unk_0C);
+            ptr->unk_1C++;
+            Math_StepToF(&ptr->unk_0C.x, 0.0f, 0.05f);
+            Math_StepToF(&ptr->unk_0C.z, 0.0f, 0.05f);
+            if (ptr->unk_0C.y < -0.5f) {
+                ptr->unk_0C.y = -0.5f;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11338.s")
+            if (i & 1) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:995]:
```suggestion
            if ((i % 2) != 0) {
```
If it matches

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 13]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+                phi_f20 = -1.0f;
+            } else {
+                phi_f20 = 1.0f;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11454.s")
+            ptr->unk_20 = Math_SinS(ptr->unk_1C * 0x7D0) * 8192.0f;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:1001]:
```suggestion
            ptr->unk_20 = Math_SinS(ptr->unk_1C * 0x7D0) * 0x2000
```
May need `(f32)0x2000`

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 14]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
+            ptr->unk_1E += (s16)(1638.0f * fabsf(Math_SinS(ptr->unk_1C * 0xBB8)) * phi_f20);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:1002]:
```suggestion
            ptr->unk_1E += (s16)(0x666 * fabsf(Math_SinS(ptr->unk_1C * 0xBB8)) * phi_f20);
```

File: src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c [hunk 15]
```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11DF0.s")
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11F6C.s")
+void EnThiefbird_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static s8 D_80C13698[] = {
+        -1, 0, -1, 1, 3, -1, 2, 5, -1, -1, 7, 8, 9, -1, -1, 10, -1, 0, 0, 0,
+    };
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c:1067]:
```suggestion
    static s8 D_80C13698[] = {
        -1, 0, -1, 1, 3, -1, 2, 5, -1, -1, 7, 8, 9, -1, -1, 10, -1,
    };
```
17 limbs, so the last 3 `0`s are pad

**Review by engineer124 — COMMENTED** [review]
Nice work! Just some comments

**Review by engineer124 — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**